### ISSUE TITLE
Issue with components install

### ISSUE THREAD

**Issue by hstanleycrow** [mnsami/composer-custom-directory-installer#14]
Hi, first, thanks for your work.
I am making some tests, and the installer works in a half, the components (jquery, jqueryui, font-awesome) has been installed on vendor directory, bootstrap, bootstrap-sass and datatables are installes as i expected, in root/resources, could you help me to find the error? if i can fix this, i will not require to use bower or anything else :)

here is my composer.json
    "require" : {
        "php": ">=5.3.0",
        "mnsami/composer-custom-directory-installer": "1.0.*",
        "components/jqueryui" : ">=1.11.4",
        "components/font-awesome" : "dev-master",
        "twbs/bootstrap-sass" : ">=3.3.5",
        "twbs/bootstrap" : "~4.0",
        "datatables/datatables" : ">=1.10.9"
    },
    "config": {
        "preferred-install": "dist"
    },
    "extra": {
        "installer-paths": {
            "public/resources/components/jqueryui": ["components/jqueryui"],
            "public/resources/components/font-awesome": ["components/font-awesome"],
            "public/resources/bootstrap": ["twbs/bootstrap"],
            "public/resources/bootstrap-sass": ["twbs/bootstrap-sass"],
            "public/resources/datatables": ["datatables/datatables"]
        }
    },  
    "minimum-stability": "dev",
    "prefer-stable": true

Thank you

**Comment by mnsami** [comment]
hello .. sorry for the late reply, but currently i'm heading off to my wedding vacations and i'll be available again after the 14th of October and I'll look into deep for all the comments you asked and pointed out.

thanks

**Comment by mnsami** [comment]
Hello @hstanleycrow 

Simply all of those packages you want to custom install them, doesn't have the right `package` type that is supported by `composer`, which are mentioned here

https://getcomposer.org/doc/04-schema.md#type

**Comment by Mariyo** [comment]
Hello,
isnt there any workaround for unsupported types? Kind of skip process...?

**Comment by mnsami** [comment]
hello @Mariyo ,

Unfortunately not at the current moment, unless you fork it and implement your own package type installer and plugin, which do the support for very specific package types.

I'm happy to discuss a solution, because it will also affect `composer` workflows.

**Comment by mnsami** [comment]
Reporter is not responding, I'm closing the ticket.

**Comment by sylfabre** [comment]
@mnsami I would like to discuss a solution :-)
I use your plugin to put jquery / jqueryui / ckeditor in a public folder

**Comment by sylfabre** [comment]
@mnsami I don't understand why you limit depending on package type and not on `installer-paths` values to select package with specific location

**Comment by mnsami** [comment]
@sylfabre hello, I would like to hear your solution, and to answer your question, each package type `composer` handles it differently, I experienced that myself and I don't want to break that handling, my aim through this plugin is to extend it.

Let me give you an example, some of the pear packages are initially a `tar` or a `zip` file which I tried before, if I used them with a different package type, they will be only downloaded and not extracted.

Second, if all package types are the same, I don't think `composer` would have differentiated them in first place.

Check also the https://packagist.org/packages/composer/installers plugin.
